numpy/_array_api/_data_type_functions.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+from ._types import Union, array, dtype
+from ._array_object import ndarray
+
+from collections.abc import Sequence
+
+import numpy as np
+
+def finfo(type: Union[dtype, array], /) -> finfo:
```

Comment:
```suggestion
def finfo(type: Union[dtype, array], /) -> np.finfo:
```
You're currently trying to return the function itself, which is in invalid type as it is an (`types.FunctionType`) instance.

Reply:
Probably makes sense to implement this instead as `class finfo` with a `__call__` method and only the attributes defined in the standard (`bits`, `eps`, `max`, `min`).

Reply:
Would It be an idea to inherit from `typing.Protocol` here? This would ensure that any object implementing aforementioned attributes would qualify as a sub-type.

Reply:
Yes, that's a great idea. It may apply in a few other places too. It's suggested for the array type annotation in https://data-apis.github.io/array-api/latest/design_topics/static_typing.html.